### PR TITLE
Update Shortest paths docs

### PR DESCRIPTION
**Pull request by nhz2**
I added some details to the Shortest paths section of the docs.

Maybe some of the specific `dijkstra_shortest_paths` info should be moved to `function dijkstra_shortest_paths` docs?

### CODE REVIEW

**Comment by nhz2** [comment]
@etiennedeg  @gdalle , could you please review this?

I would like to write an edge betweenness function using the `dijkstra_shortest_paths` in  `Graphs.jl`, but its outputs are currently poorly documented.

File: docs/src/first_steps/paths_traversal.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ The package also includes uniform random walks and self avoiding walks with the
 The following properties always hold for shortest path algorithms implemented here:
 
 - The distance from a vertex to itself is always `0`.
-- The distance between two vertices with no connecting edge is always `Inf`.
+- The distance between two vertices with no connecting edge is always `Inf` or `typemax(Int64)`.
```
Comment (etiennedeg) [inline docs/src/first_steps/paths_traversal.md:34]:
```suggestion
- The distance between two vertices with no connecting edge is always `Inf` or `typemax(eltype(distmx))`.
```

**Review by etiennedeg — COMMENTED** [review]
Thanks for your contribution, I think we should also document this in [Algorithms API](https://juliagraphs.org/Graphs.jl/dev/algorithms/shortestpaths/), this is actually less documented than the [Paths and traversal](https://juliagraphs.org/Graphs.jl/dev/first_steps/paths_traversal/) section.
As a side note, we should also add accessors for the `AbstractPath` object, we should not need to rely on internal fields.

**Comment by nhz2** [comment]
> Thanks for your contribution, I think we should also document this in [Algorithms API](https://juliagraphs.org/Graphs.jl/dev/algorithms/shortestpaths/), this is actually less documented than the [Paths and traversal](https://juliagraphs.org/Graphs.jl/dev/first_steps/paths_traversal/) section. As a side note, we should also add accessors for the `AbstractPath` object, we should not need to rely on internal fields.

Thanks for the review, I moved the `dijkstra_shortest_paths` specific info to the `dijkstra_shortest_paths` docstring,
and added links to the referenced functions in [Paths and traversal](https://juliagraphs.org/Graphs.jl/dev/first_steps/paths_traversal/).

**Review by etiennedeg — APPROVED** [review]

**Comment by gdalle** [comment]
Thanks @etiennedeg

**Pull request merged**